NodeJS-HW3/src/routers/userRouter.ts

```diff
@@ -0,0 +1,58 @@
+import { Router, Response } from 'express';
+
+import { userSchema, updateSchema } from '../config/schemas';
+import { validateSchema } from '../utils/helpers';
+import UserService from '../services/userService';
+
+const router = Router();
+
+const userService = new UserService();
+
+router.param('id', async (_, res: Response, next: () => void, id) => {
+  const user = await userService.findUserById(id);
+  if (!user)
+    res.status(404).json({ message: `User with id=${id} was not found!` });
+  else {
+    res.locals.user = user;
+    next();
+  }
+});
+
+router.get('/', async (req, res) => {
+  const { loginSubstring, limit } = req.query;
+  const users = await userService.getExistingUsers();
+
+  if (loginSubstring && limit) {
+    const filteredUsers = users
+      .filter((user) =>
+        user.getDataValue('login').includes(`${loginSubstring}`),
+      )
+      .slice(0, +limit);
```

Comment:
plz add a service method that does this filtering. You should only call service methods for any kind of logic.